main.go

```diff
@@ -73,10 +68,35 @@ func main() {
 
 	// Serve metrics
 	http.Handle(cfg.MetricPath, promhttp.Handler())
-	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
-		w.Write([]byte(indexContent))
-	})
 
-	log.Println("Listening on", cfg.ListenAddress)
-	log.Fatal(http.ListenAndServe(cfg.ListenAddress, nil))
+	if cfg.MetricPath != "/" {
+		landingConfig := web.LandingConfig{
+			Name:        "Squid Exporter",
+			Description: "Prometheus exporter for Squid caching proxy servers",
+			HeaderColor: "#15a5be",
+			Version:     version.Info(),
+			Links: []web.LandingLinks{
+				{
+					Address: cfg.MetricPath,
+					Text:    "Metrics",
+				},
+			},
+		}
+		landingPage, err := web.NewLandingPage(landingConfig)
+		if err != nil {
+			log.Fatal(err)
+		}
+		http.Handle("/", landingPage)
+	}
+
+	systemdSocket := false
+	toolkitFlags := &web.FlagConfig{
+		WebListenAddresses: &[]string{cfg.ListenAddress},
+		WebSystemdSocket:   &systemdSocket,
+		WebConfigFile:      &cfg.WebConfigFile,
```

Comment:
Is this optional? If user does not provide this flag (ie. empty string is the default), does this still work as expected?

Reply:
Yes, it works as expected. Note that setting `toolkitFlags` in such a way is only necessary because it is not currently feasible to do something like this:
```go
import "github.com/prometheus/exporter-toolkit/web/kingpinflag"

...

toolkitFlags = kingpinflag.AddFlags(kingpin.CommandLine, ":9301")
```
(so long as squid-exporter is still using the standard lib flag package)